.github/actions/update-extension-provider-images/action.yaml

```diff
@@ -0,0 +1,64 @@
+#
+# SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+name: 'Update Extension Provider Images'
+description: 'Checks for new image versions, updates images.yaml, and generates release notes.'
+
+inputs:
+  images-yaml-path:
+    description: 'Path to the images.yaml file.'
+    required: false
+    default: 'imagevector/images.yaml'
+  release-notes-path:
+    description: 'Path where the release-notes.md file will be generated.'
+    required: false
+    default: 'release-notes.md'
+
+outputs:
+  summary:
+    description: "A summary of the image updates performed by the script."
+    value: ${{ steps.run_script_step.outputs.summary }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Validate inputs
+      shell: bash
+      run: |
+        if [[ ! -f "${{ inputs.images-yaml-path }}" ]]; then
+          echo "Error: images.yaml file not found at ${{ inputs.images-yaml-path }}"
+          exit 1
+        fi
+
+    - name: Install Gardener GHA Libs
+      uses: ./.github/actions/install-gardener-gha-libs
+
+    - name: Install additional Python dependencies
+      shell: bash
+      run: pip install -r ${{ github.action_path }}/requirements.txt
+
+    - name: Run image update script
+      id: run_script_step
+      shell: bash
+      run: |
+        set -euo pipefail
+
+        summary=$(python3 ${{ github.action_path }}/update-images.py \
+          --images-yaml-path ${{ inputs.images-yaml-path }} \
+          --release-notes-path ${{ inputs.release-notes-path }})
+        
+        exit_code=$?
```

Comment:
as you set `-euo pipefail` above, this line will not be reached in case of exitcode !== 0. Which is fine (I suggest to simply drop unnecessary boilerplate)